todo_app/task/api/views.py

```diff
@@ -1,59 +1,75 @@
 from task.serializers import TaskSerializer
 
 
-class TaskList(LoginRequiredMixin, APIView):
+class TaskList(APIView):
+    permission_classes = [IsAuthenticated]
+    serializer_setter = TaskSerializer
+
     def get(self, request):
-        user_profile = request.user.userprofile
+        user_profile = request.user
 
-        if user_profile.user.is_superuser:
+        if user_profile.is_superuser:
             tasks = Task.objects.all()
         else:
             tasks = Task.objects.filter(user=user_profile)
 
         tasks = tasks.order_by("status", "due_date")
-        serializer = TaskSerializer(tasks, many=True)
+
+        serializer_class = self.serializer_setter.get_serializer(request.user)
+        serializer = serializer_class(tasks, many=True)
         return Response(serializer.data)
 
     def post(self, request):
-        serializer = TaskSerializer(data=request.data)
+        serializer_class = self.serializer_setter.get_serializer(request.user)
+        serializer = serializer_class(data=request.data)
+
         if serializer.is_valid():
-            if not request.user.userprofile.user.is_superuser:
-                serializer.save(user=request.user.userprofile)
+            if not request.user.is_superuser:
+                serializer.save(user=request.user)
             else:
                 serializer.save()
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-class SpecficTask(LoginRequiredMixin, APIView):
+class SpecficTask(APIView):
+    permission_classes = [IsAuthenticated]
+    serializer_setter = TaskSerializer
+
     def get(self, request, pk):
         try:
             task = Task.objects.get(task_id=pk)
-            if (task.user.user.username == request.user.username) or (
-                request.user.userprofile.user.is_superuser
+            if (task.user.username == request.user.username) or (
+                request.user.is_superuser
             ):
-                serializer = TaskSerializer(task, many=False)
+                serializer_class = self.serializer_setter.get_serializer(request.user)
+                serializer = serializer_class(task, many=False)
+
                 return Response(serializer.data)
             return Response(status=status.HTTP_403_FORBIDDEN)
         except Task.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
 
-class UpdateTask(LoginRequiredMixin, APIView):
+class UpdateTask(APIView):
+    permission_classes = [IsAuthenticated]
+    serializer_setter = TaskSerializer
+
     def get(self, request, pk):
         try:
             task = Task.objects.get(task_id=pk)
-            if (task.user.user.username == request.user.username) or (
-                request.user.userprofile.user.is_superuser
+            if (task.user.username == request.user.username) or (
+                request.user.is_superuser
             ):
```

Comment:
can we achieve this using custom Django permission?

Reply:
Let me have a look

Reply:
Added